semana-16/projeto/to_do_list/src/data/selectDelayedTasks.ts

```diff
@@ -0,0 +1,18 @@
+import { connection } from ".."
+import { currDate } from "../handleDate";
+
+export const selectDelayedTasks = async (): Promise<any[]> => {
+  return await connection("ToDoListTask")
+    .join("ToDoListUser", "ToDoListTask.creator_user_id", "ToDoListUser.id")
+    .select(
+      "ToDoListTask.id as taskId",
+      "ToDoListTask.title as title",
+      "ToDoListTask.description as description",
+      "ToDoListTask.status as status",
+      "ToDoListTask.limit_date as limitDate",
+      "ToDoListUser.id as creatorUserId",
+      "ToDoListUser.nickname as creatorUserNickname"
+    )
+    .where("ToDoListTask.limit_date", "<", currDate)
+    .andWhereNot("ToDoListTask.status", "done")
+}
```

Comment:
ainda não me sinto tão seguro usando o query build do knex.. nesse caso acho que usaria o raw.
você acha que seria indicado usar só um tipo de consulta ou poderia mistura query builds e raw? fiquei na dúvida por que tem coisas que são mais simples com uma do que com outras.

Reply:
Acho q é bom saber fazer de todas as maneiras. Mas, não sei pq, me identifiquei mais com o query builder. Me parece mais clean, mais organizado.
E to smp com o manual do knex do lado pra lembrar como se faz tudo e tentar entender melhor.